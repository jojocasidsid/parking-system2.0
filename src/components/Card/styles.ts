import styled from '@emotion/styled'

export const CardWrapper = styled('div')`
	display: flex;
	justify-content: flex-start;
	align-items: center;
	padding: 20px;
	flex-direction: column;
	border-radius: 30px;
	border: 2px solid ${({ theme }) => theme.palette.secondary.main};
`
export const StyledSlotHeader = styled('div')`
	display: flex;
	justify-content: space-around;
	align-items: center;
`
