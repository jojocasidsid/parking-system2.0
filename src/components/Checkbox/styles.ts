import styled from '@emotion/styled'

export const StyledRoot = styled('div')`
	padding: 18px;
`
