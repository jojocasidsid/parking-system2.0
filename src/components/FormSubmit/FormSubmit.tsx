import React from 'react'
import { Button, Grid, CircularProgress } from '@mui/material'

interface IProps {
	submitting: boolean
	handleClose: () => void
	actionText: string
}

const FormSubmit = ({ submitting, handleClose, actionText }: IProps) => (
	<Grid item container xs={12}>
		<Grid item xs={8} />
		<Grid item xs={4} container columnSpacing={1.5} justifyContent='flex-end'>
			<Grid item>
				<Button
					variant='outlined'
					color='secondary'
					onClick={handleClose}
					disabled={submitting}
				>
					Cancel
				</Button>
			</Grid>
			<Grid item>
				<Button variant='contained' type='submit' disabled={submitting}>
					{submitting ? <CircularProgress size={24.5} /> : actionText}
				</Button>
			</Grid>
		</Grid>
	</Grid>
)

export default FormSubmit
