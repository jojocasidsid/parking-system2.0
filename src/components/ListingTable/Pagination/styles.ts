import styled from '@emotion/styled'
import { Typography } from '@mui/material'

export interface IPageInputProps {
	error?: string
}

export const PageInput = styled.input<IPageInputProps>`
	border-radius: 2px 0px 0px 2px;
	padding: 7px 2px 7px 0px;
	text-align: center;
	width: 100%;
	font-style: normal;
	font-size: 14px;
	line-height: 16px;
	font-weight: 500;
	font-stretch: normal;
	height: 28px;

	&::-webkit-outer-spin-button,
	&::-webkit-inner-spin-button {
		-webkit-appearance: none;
		margin: 0;
	}

	&:active,
	&:focus {
		border: 1px solid ${({ theme }) => theme.palette.primary.main};
		outline: none;
		box-shadow: none;
	}

	&:disabled {
		background-color: ${({ theme }) => theme.palette.text.secondary};
		pointer-events: none;
	}

	border: 1px solid
		${({ theme, error }) => {
			if (error) {
				return theme?.palette.error.main
			}
			return theme.palette.text.secondary
		}};
`

export const StyledTypography = styled(Typography)`
	color: #303643b3;
	font-weight: bold;
	font-size: 12px;
`
