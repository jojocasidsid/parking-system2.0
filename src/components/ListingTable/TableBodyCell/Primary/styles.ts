import styled from '@emotion/styled'
import Typography from '@mui/material/Typography'

export const StyledTypography = styled(Typography)`
	color: #303643b3;
	font-weight: bold;
`
