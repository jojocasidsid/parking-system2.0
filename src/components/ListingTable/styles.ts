import styled from '@emotion/styled'
import { Typography } from '@mui/material'

export const StyledTypography = styled(Typography)`
	color: #30364380;
	font-weight: bold;
	font-size: 14px;
`
