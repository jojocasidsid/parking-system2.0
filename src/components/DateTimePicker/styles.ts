import styled from '@emotion/styled'
import { InputLabel, TextField } from '@mui/material'
import { Theme } from '@mui/material/styles'

interface ITheme {}
interface IInput extends ITheme {
	errors?: boolean
	theme?: Theme
}

export const StyledRoot = styled('div')`
	display: block;
`

export const StyledLabel = styled(InputLabel)`
	text-align: 'left';
	letter-spacing: 0.12px;
	padding-bottom: 8px;

	font-style: normal;
	font-size: 12px;
	line-height: 14px;
	font-weight: 600;
	font-stretch: normal;
	white-space: normal;
	//has error
	color: ${({ theme, errors }: IInput) => {
		if (errors) {
			return theme?.palette.error.main
		}
		return theme?.palette.text.primary
	}};
	& span {
		color: ${({ theme }) => theme.palette.error.main};
	}

	display: block;
`

export const StyledInput = styled(TextField)`
	input {
		padding: 8px 16px 8px 16px;
		font-style: normal;
		font-size: 14px;
		font-weight: 500;
		font-stretch: normal;
		color: ${({ theme }) => theme.palette.text.primary};
	}

	fieldset {
		border: 1px solid ${({ theme }) => theme.palette.text.secondary};
		border-radius: 2px 0px 0px 2px;
		border-color: ${({ theme, errors }: IInput) =>
			errors
				? theme?.palette.error.main
				: theme?.palette.text.secondary} !important;
	}

	input.Mui-disabled {
		background-color: ${({ theme }) => theme.palette.text.secondary};
		pointer-events: none;
	}

	input.MuiInputBase-inputAdornedStart {
		padding-left: 0px;
	}

	input.MuiInputBase-inputAdornedEnd {
		padding-right: 8px;
	}

	.MuiFormHelperText-root {
		color: ${({ theme }) => theme.palette.error.main};
		margin-left: 0;
	}
`
