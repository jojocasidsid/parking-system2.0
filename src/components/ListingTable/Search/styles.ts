import styled from '@emotion/styled'
import { InputLabel, TextField } from '@mui/material'
import { Theme } from '@mui/material/styles'

interface IInput {
	error?: boolean
	theme?: Theme
	small?: boolean
}

export const StyledRoot = styled('div')`
	display: block;
`

export const StyledLabel = styled(InputLabel)`
	text-align: 'left';
	letter-spacing: 0.12px;
	padding-bottom: 8px;

	font-style: normal;
	font-size: 12px;
	line-height: 14px;
	font-weight: 600;
	font-stretch: normal;
	white-space: normal;
	//has error
	color: ${({ theme, error }: IInput) => {
		if (error) {
			return theme?.palette.error.main
		}
		return theme?.palette.text.primary
	}};
	& span {
		color: ${({ theme }) => theme.palette.error.main};
	}

	display: block;
`

export const StyledInput = styled(TextField)`
	input {
		padding: ${({ small }: IInput) =>
			small ? '2px 8px 2px 8px' : '8px 12px 8px 16px'};
		font-style: normal;
		font-size: 14px;
		font-weight: 500;
		font-stretch: normal;
		color: ${({ theme }) => theme?.palette.text.primary};
	}

	fieldset {
		border: 1px solid ${({ theme }) => theme.palette.text.secondary};
		border-radius: 2px 0px 0px 2px;
	}

	input.Mui-disabled {
		background-color: ${({ theme }) => theme.palette.text.secondary};
		pointer-events: none;
	}

	input.MuiInputBase-inputAdornedStart {
		padding-left: 0px !important;
	}

	input.MuiInputBase-inputAdornedEnd {
		padding-right: 0px;
	}

	.MuiInputBase-adornedEnd div.MuiInputAdornment-root {
		background-color: ${({ theme }) => theme.palette.primary.contrastText};
		padding: 16px 16px 16px 16px;
		padding-right: 0px !important;
	}

	.MuiInputBase-adornedStart {
		padding-left: 0px !important;
	}

	.MuiInputBase-adornedStart .MuiInputAdornment-root {
		background-color: ${({ theme }) => theme.palette.primary.contrastText};
		padding: 16px 16px 16px 16px;
	}

	.MuiFormHelperText-root {
		margin-left: 0;
	}
`
