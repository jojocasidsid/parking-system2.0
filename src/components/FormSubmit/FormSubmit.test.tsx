import React from 'react'

// packages
import renderer from 'react-test-renderer'
import { ThemeProvider } from '@mui/material/styles'

// config
import theme from 'config/theme'

import FormSubmit from './FormSubmit'

describe('FormSubmit Component', () => {
	it('should render component without crashing on loading', () => {
		const tree = renderer.create(
			<ThemeProvider theme={theme}>
				<FormSubmit submitting handleClose={() => {}} actionText='Park' />
			</ThemeProvider>
		)

		expect(tree).toMatchSnapshot()
	})

	it('should render component without crashing', () => {
		const tree = renderer.create(
			<ThemeProvider theme={theme}>
				<FormSubmit submitting={false} handleClose={() => {}} actionText='Unpark' />
			</ThemeProvider>
		)

		expect(tree).toMatchSnapshot()
	})
})
