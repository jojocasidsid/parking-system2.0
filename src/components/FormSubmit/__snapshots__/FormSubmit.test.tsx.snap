// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormSubmit Component should render component without crashing on loading 1`] = `
<div
  className="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-grid-xs-12 css-spompt-MuiGrid-root"
>
  <div
    className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-8 css-vok43g-MuiGrid-root"
  />
  <div
    className="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-grid-xs-4 css-1nh5xci-MuiGrid-root"
  >
    <div
      className="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
    >
      <button
        className="MuiButton-root MuiButton-outlined MuiButton-outlinedSecondary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButtonBase-root Mui-disabled  css-evvnld-MuiButtonBase-root-MuiButton-root"
        disabled={true}
        onBlur={[Function]}
        onClick={[Function]}
        onContextMenu={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={-1}
        type="button"
      >
        Cancel
      </button>
    </div>
    <div
      className="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
    >
      <button
        className="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root Mui-disabled  css-1x63eaz-MuiButtonBase-root-MuiButton-root"
        disabled={true}
        onBlur={[Function]}
        onContextMenu={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={-1}
        type="submit"
      >
        <span
          className="MuiCircularProgress-root MuiCircularProgress-indeterminate MuiCircularProgress-colorPrimary css-19r7jhb-MuiCircularProgress-root"
          role="progressbar"
          style={
            Object {
              "height": 24.5,
              "width": 24.5,
            }
          }
        >
          <svg
            className="MuiCircularProgress-svg css-1idz92c-MuiCircularProgress-svg"
            viewBox="22 22 44 44"
          >
            <circle
              className="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate css-176wh8e-MuiCircularProgress-circle"
              cx={44}
              cy={44}
              fill="none"
              r={20.2}
              strokeWidth={3.6}
              style={Object {}}
            />
          </svg>
        </span>
      </button>
    </div>
  </div>
</div>
`;

exports[`FormSubmit Component should render component without crashing on loading 2`] = `
<div
  className="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-grid-xs-12 css-spompt-MuiGrid-root"
>
  <div
    className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-8 css-vok43g-MuiGrid-root"
  />
  <div
    className="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-grid-xs-4 css-1nh5xci-MuiGrid-root"
  >
    <div
      className="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
    >
      <button
        className="MuiButton-root MuiButton-outlined MuiButton-outlinedSecondary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButtonBase-root  css-evvnld-MuiButtonBase-root-MuiButton-root"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onContextMenu={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        Cancel
        <span
          className="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
    </div>
    <div
      className="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
    >
      <button
        className="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root  css-1x63eaz-MuiButtonBase-root-MuiButton-root"
        disabled={false}
        onBlur={[Function]}
        onContextMenu={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="submit"
      >
        Park
        <span
          className="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
    </div>
  </div>
</div>
`;
